master/internal/stream/authz_iface.go

```diff
@@ -0,0 +1,21 @@
+package stream
+
+import (
+	"context"
+
+	"github.com/determined-ai/determined/master/internal/authz"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// StreamAuthZ is the interface for streaming authorization.
+type StreamAuthZ interface {
+	// GetTrialStreamableScopes returns a AccessScopeSet where the user has permission to view trials.
+	GetTrialStreamableScopes(ctx context.Context, curUser model.User) (model.AccessScopeSet, error)
+
+	// WaitForPermissionChange returns an error whenever
+	// a change that modifies permission scopes are detected.
```

Comment:
"block until you return an error" is an odd api.

If it can't fail, it shouldn't return an error.  If it can fail, it should probably distinguish "I failed" from "yup there's been a permission change".

Reply:
I also agree, this feels weird to make a call to authz & have it be a blocking call. 

The intention was to allow OSS to short circuit through any "wait for permission change"-related logic because it doesn't need to be monitoring for changes.  

Instead this call could return a *pq.Listener already listening to the correct channel or it could return the function that will perform the wait+preemption.